src/Html2OpenXml/HtmlConverter.cs

```diff
@@ -120,36 +120,75 @@ public IList<OpenXmlCompositeElement> Parse(String html)
 			return paragraphs;
 		}
 
-        /// <summary>
+		/// <summary>
 		/// Start the parse processing and append the converted paragraphs into the Body of the document.
 		/// </summary>
-        public void ParseHtml(String html)
-        {
-            // This method exists because we may ensure the SectionProperties remains the last element of the body.
-            // It's mandatory when dealing with page orientation
-
-            var paragraphs = Parse(html);
+		public void ParseHtml(String html) {
+			// This method exists because we may ensure the SectionProperties remains the last element of the body.
+			// It's mandatory when dealing with page orientation
+
+			var paragraphs = Parse(html);
+
+			Body body = mainPart.Document.Body;
+			SectionProperties bodySectionProperties = body.GetLastChild<SectionProperties>();
+			bodySectionProperties?.Remove();
+			var firstparaSectionProps = body.Descendants<SectionProperties>().FirstOrDefault();
+			var lastSectionIsPaginated = false;
+			PageNumberType pageNumType = null;
+			FooterReference footerRef = null;
+			//if we find a pageNumType and the last section is paginated the most likely we need to pass it on to the first added para.
+			if (!(bodySectionProperties is null)) {
+				pageNumType = bodySectionProperties.GetFirstChild<PageNumberType>();
+				pageNumType?.Remove();
+				footerRef = bodySectionProperties.GetFirstChild<FooterReference>();
+			}
 
-            Body body = mainPart.Document.Body;
-			SectionProperties sectionProperties = body.GetLastChild<SectionProperties>();
-			sectionProperties?.Remove();
+			if (firstparaSectionProps != null && bodySectionProperties.GetFirstChild<FooterReference>() != null)
+				lastSectionIsPaginated = true;
 
 			for (int i = 0; i < paragraphs.Count; i++)
-                body.Append(paragraphs[i]);
+				body.Append(paragraphs[i]);
 
 			// Push the sectionProperties as the last element of the Body
 			// (required by OpenXml schema to avoid the bad formatting of the document)
-			body.Append(HtmlConverter.ChangePageOrientation(currentOrientation));
-			
+			var newOrientationSettings = HtmlConverter.ChangePageOrientation(currentOrientation);
+			if (bodySectionProperties is null) {
+				body.Append(newOrientationSettings);
+				bodySectionProperties = newOrientationSettings;
+            } else {
+                var firstAddedParaSectProps = paragraphs.First().Descendants<SectionProperties>().First();

```

Comment:
```suggestion
                var firstAddedParaSectProps = paragraphs.First(p => p.Descendants<SectionProperties>().Any()).Descendants<SectionProperties>().First();
```
The first paragraph in the array might not have a SectionProperties but others might. Also should we use `FirstOrDefault`?